### DISCUSSION TITLE
Allow configuring `suppressions/unused` rule fix

### DISCUSSION THREAD

**Discussion by kymckay** [biomejs/biome#3629] [Ideas]
Currently the `suppressions/unused` rule flags as an unsafe fix. I'd love to be able to configure the rule fix to be considered safe so that these automatically get cleaned up.

For clarity: it seems like this rule specifically is not configurable in `biome.json`.

**Comment by ematipico** [comment]
There's some misunderstanding.

`suppression/unused` isn't the name of a rule, it's the category of the diagnostic.

That's not a lint rule, and I believe there's no code action (fix) associated with it.

Given that, what kind of resolution/proposal are you looking for? Can you be more specific about your idea and provide more examples?

**Reply by kymckay** [reply]
Fast response! Yep, can explain the misunderstanding too 😄 This
>The icon ⚠ indicates that the rule provides a code action (fix) that is unsafe to apply.

in the docs made me think there was an unsafe fix suggested when I saw the same symbol in the CLI.

I'm in the process of migrating a codebase over from ESLint and as part of that some of the old suppression statements (which I've just replaced in bulk) no longer apply. My thinking was that it'd be nice to configure Biome to strip such unused suppression statements out, because it's already doing the work of detecting that they're unused.

**Comment by GRA0007** [comment]
Adding to this, I'd like at least the ability to see a list of all unused suppression comments in the codebase. Using `biome lint --summary` shows that I have 22 with the rule name `suppressions/unused`, but I can't find any option to filter the output so I get a list of files with these warnings. We have a very large codebase with lots of other warnings at the moment, and when we change our Biome config or refactor code, we often get unused suppression comments.

**Reply by ematipico** [reply]
Yes, that's on my radar too.

Having a better suppression system inspired by `elm-review`.

We'll get to that eventually

**Comment by ipanasenko** [comment]
It would be nice for `suppressions/unused` to behave similar to `@ts-expect-error`, e.g. if there is an unused suppession - fail linting. With autofix unused suppressions can be removed, for example

**Comment by maelp** [comment]
There are some case though where we don't want that, and would like to suppress the `suppressions/unused` errors from linting

in my case:

I'm using the Zed editor

and there's a bug in the Zed <-> Biome integration (it doesn't find the "closest" biome.jsonc file to the current file it edits, so it applies the wrong linting rules)

but because Zed is configured to do some auto-formatting on save, it keeps applying rules (to try to "fix" things) that it hallucinates (it doesn't have the proper biome.jsonc file)

So I have to explicitely add a `// biome-ignore` comment on the rules that it hallucinates to prevent it from trying to "autofix it" on save

but obviously then when my CI/CD does a linting, it finds the `// biome-ignore` unnecessary... but I'd still want to keep it (as a fix of the Zed bug)

**Comment by markedwards** [comment]
I  have a different form of request for configuring this -- I want these to be considered to be errors so they must be removed in order to pass lint.  And I don't want to have to declare all warnings to be errors to achieve this.  Is that on the roadmap?

**Reply by dyc3** [reply]
They already cause biome to return a non-zero exit code, and thus not pass linting. Or do you not consider warnings to be blocking?
